### PR TITLE
Fix isLayoutSupported in amp-experiment

### PR DESCRIPTION
**Pull request by muxin**

### CODE REVIEW

File: extensions/amp-experiment/0.1/amp-experiment.js [hunk 1]
```diff
@@ -28,8 +28,8 @@ const ATTR_PREFIX = 'amp-x-';
 export class AmpExperiment extends AMP.BaseElement {
 
   /** @override */
-  isLayoutSupported(unusedLayout) {
-    return true;
+  isLayoutSupported(layout) {
+    return layout == Layout.NODISPLAY || layout == Layout.CONTAINER;
```
Comment (lannka) [inline extensions/amp-experiment/0.1/amp-experiment.js]: could you explain a little bit why is this change needed?
This was to mimic `amp-analytics`'s behavior.

If we want to make sure it's never displayed, we can probably explicitly set the element to hidden in `buildCallback`.

Comment (dvoytenko) [inline extensions/amp-experiment/0.1/amp-experiment.js]: Setting it to hidden in `buildCallback` is too late - it might have already been visible for a long time before the `buildCallback` had a chance to download and upgrade. Instead, @muxin explicitly set it to `display: none;` in another PR per my request. This is just a follow up to make sure users don't even try to set width/height and other layout attributes on this element.

**Comment by dvoytenko** [comment]
LGTM

File: extensions/amp-experiment/0.1/amp-experiment.js (hunk 1 above)
Comment (lannka) [inline extensions/amp-experiment/0.1/amp-experiment.js]: how about `amp-analytics`? does it have the same problem?

**Comment by dvoytenko** [comment]
@lannka Yes, ideally we'd do the same there as well.

File: extensions/amp-experiment/0.1/amp-experiment.js (hunk 1 above)
Comment (muxin) [inline extensions/amp-experiment/0.1/amp-experiment.js]: @dvoytenko

Comment (dvoytenko) [inline extensions/amp-experiment/0.1/amp-experiment.js]: Responded below. But yes, amp-analytics needs to do the same.
